File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/extras/PlayerInfoSwipeDetector.java
package com.devingotaswitch.rankings.extras;

import android.view.MotionEvent;
import android.view.View;

import com.devingotaswitch.rankings.PlayerInfo;

public class PlayerInfoSwipeDetector implements View.OnTouchListener {

    private static final int MIN_DISTANCE = 450;
    private float downX, upX;
    private PlayerInfo hold;

    public PlayerInfoSwipeDetector(PlayerInfo playerInfo) {
        hold = playerInfo;
    }

    private final void onRightToLeftSwipe() {
        hold.swipeRightToLeft();
    }

    private void onLeftToRightSwipe() {
        hold.swipeLeftToRight();
    }

    public boolean onTouch(View v, MotionEvent event) {
        switch (event.getAction()) {
            case MotionEvent.ACTION_DOWN: {
                downX = event.getX();
            }
            case MotionEvent.ACTION_UP: {
                upX = event.getX();
                float deltaX = downX - upX;
                if (Math.abs(deltaX) > MIN_DISTANCE) {
                    if (deltaX < 0) {
                        this.onLeftToRightSwipe();
                        return true;
                    }
                    if (deltaX > 0) {
                        this.onRightToLeftSwipe();
                        return true;
                    }
                }
            }
        }
        return false;
    }
}

File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/extras/PlayerInfoSwipeDetector.kt
package com.devingotaswitch.rankings.extras

import android.view.MotionEvent
import android.view.View
import com.devingotaswitch.rankings.PlayerInfo
import kotlin.math.abs

class PlayerInfoSwipeDetector(private val hold: PlayerInfo) : View.OnTouchListener {
    private var downX = 0f
    private fun onRightToLeftSwipe() {
        hold.swipeRightToLeft()
    }

    private fun onLeftToRightSwipe() {
        hold.swipeLeftToRight()
    }

    override fun onTouch(v: View, event: MotionEvent): Boolean {
        when (event.action) {
            MotionEvent.ACTION_DOWN -> {
                run { downX = event.x }
                run {
                    val upX = event.x
                    val deltaX = downX - upX
                    if (abs(deltaX) > MIN_DISTANCE) {
                        if (deltaX < 0) {
                            this.onLeftToRightSwipe()
                            return true
                        }
                        if (deltaX > 0) {
                            this.onRightToLeftSwipe()
                            return true
                        }
                    }
                }
            }
            MotionEvent.ACTION_UP -> {
                val upX = event.x
                val deltaX = downX - upX
                if (abs(deltaX) > MIN_DISTANCE) {
                    if (deltaX < 0) {
                        onLeftToRightSwipe()
                        return true
                    }
                    if (deltaX > 0) {
                        onRightToLeftSwipe()
                        return true
                    }
                }
            }
        }
        return false
    }

    companion object {
        private const val MIN_DISTANCE = 450
    }
}